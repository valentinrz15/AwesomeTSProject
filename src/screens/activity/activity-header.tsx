import React, { FunctionComponent } from 'react';
import { View, TouchableOpacity, Text } from 'react-native';
import styles from './styles';

interface ActivityHeaderProps {
  titleOne: string;
  titleTwo: string;
  actionOne?: () => void;
  actionTwo?: () => void;
}

export const ActivityHeader: FunctionComponent<ActivityHeaderProps> = ({
  titleOne,
  titleTwo,
  actionOne,
  actionTwo,
}) => {
  return (
    <View style={styles.header}>
      <TouchableOpacity onPress={actionOne} style={styles.headerButtons}>
        <Text style={styles.headerTitles}>{titleOne}</Text>
      </TouchableOpacity>
      <TouchableOpacity onPress={actionTwo} style={styles.headerButtons}>
        <Text style={styles.headerTitles}>{titleTwo}</Text>
      </TouchableOpacity>
    </View>
  );
};
