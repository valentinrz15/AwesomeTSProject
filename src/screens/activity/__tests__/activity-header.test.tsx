import React from 'react';
import renderer from 'react-test-renderer';
import { ActivityHeader } from '../activity-header';

describe('activity header test', () => {
  it('should render activity header titles', () => {
    const { root, toJSON } = renderer.create(
      <ActivityHeader titleOne="Close" titleTwo="Filters" />,
    );
    const { titleOne, titleTwo } = root.props;

    expect(titleOne).toBe('Close');
    expect(titleTwo).toBe('Filters');
    expect(toJSON()).toMatchSnapshot();
  });
});
