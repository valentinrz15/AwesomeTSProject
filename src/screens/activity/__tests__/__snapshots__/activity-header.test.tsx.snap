// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`activity header test should render activity header titles 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 20,
      "paddingVertical": 10,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#000",
        "borderColor": "#ccc",
        "borderRadius": 5,
        "borderWidth": 0.5,
        "opacity": 1,
        "paddingHorizontal": 5,
        "paddingVertical": 3,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFF",
          "fontSize": 18,
        }
      }
    >
      Close
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#000",
        "borderColor": "#ccc",
        "borderRadius": 5,
        "borderWidth": 0.5,
        "opacity": 1,
        "paddingHorizontal": 5,
        "paddingVertical": 3,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFF",
          "fontSize": 18,
        }
      }
    >
      Filters
    </Text>
  </View>
</View>
`;
